tests/index_selector.json

```diff
@@ -85,6 +113,26 @@
       "selector": "$[01]",
       "invalid_selector": true
     },
+    {
+      "name": "decimal",
+      "selector": "$[1.0]",
+      "invalid_selector": true
+    },
+    {
+      "name": "plus",
+      "selector": "$[+1]",
```

Comment:
Why is this invalid ? in Java, this is valid.
```java
  private def `int`[_: P]: P[Int] = P("0" | (CharIn("+\\-").? ~ `DIGIT1` ~ `DIGIT`.rep)).!
    .map(_.toLong)
    .filter(value => MIN_INTEGER < value && value < MAX_INTEGER)
    .map(num => Math.min(Int.MaxValue, Math.max(Int.MinValue, num)).toInt)
```

Reply:
The standard does not allow `+` for integers.
```
index-selector      = int                        ; decimal integer

int                 = "0" /
                      (["-"] DIGIT1 *DIGIT)      ; - optional
DIGIT1              = %x31-39                    ; 1-9 non-zero digit
```
https://ietf-wg-jsonpath.github.io/draft-ietf-jsonpath-base/draft-ietf-jsonpath-base.html#name-syntax-5

Reply:
Oh, I see. It seems I missed this, but I think `+` can also be optional, will update my implementation, thanks